go/connection_test.go

```diff
@@ -287,6 +287,140 @@ func TestInterpolateParamsUint64(t *testing.T) {
 	assert.Equal(t, q, "SELECT 42")
 }
 
+func TestInterpolateParamsDateOnly(t *testing.T) {
+	testTime, err := time.Parse(time.DateOnly, "2024-07-01")
+	assert.Nil(t, err)
+
+	c := createTestConnection(t)
+	q, err := c.interpolateParams("SELECT ?", []driver.Value{testTime})
+	assert.Nil(t, err)
+	assert.Equal(t, "SELECT '2024-07-01 00:00:00'", q)
+}
+
+func TestInterpolateParamsTime(t *testing.T) {
+	testTime, err := time.Parse(time.RFC3339, "2024-07-01T01:02:03Z")
+	assert.Nil(t, err)
+
+	c := createTestConnection(t)
+	q, err := c.interpolateParams("SELECT ?", []driver.Value{testTime})
+	assert.Nil(t, err)
+	assert.Equal(t, "SELECT '2024-07-01 01:02:03'", q)
+}
+
+func TestInterpolateParamsTimeMicro(t *testing.T) {
+	testTimeMicro, err := time.Parse(time.RFC3339, "2024-07-02T01:02:03Z")
+	assert.Nil(t, err)
+	testTimeMicro = testTimeMicro.Add(time.Microsecond * 123456)
+
+	c := createTestConnection(t)
+	q, err := c.interpolateParams("SELECT ?", []driver.Value{testTimeMicro})
+	assert.Nil(t, err)
+	assert.Equal(t, "SELECT '2024-07-02 01:02:03.123456'", q)
+}
+
+func TestBuildExecutionParams(t *testing.T) {
+	testTime, err := time.Parse(time.RFC3339, "2024-07-01T00:00:00Z")
+	assert.Nil(t, err)
+	testTimeMicro, err := time.Parse(time.RFC3339, "2024-07-02T01:02:03Z")
+	assert.Nil(t, err)
+	testTimeMicro = testTimeMicro.Add(time.Microsecond * 123456)
+
+	testCases := []struct {
+		name        string
+		inputArgs   []driver.Value
+		expectedErr error
+		expected    []*string
+	}{
+		{
+			name:        "No arguments",
+			inputArgs:   []driver.Value{},
+			expectedErr: nil,
+			expected:    []*string{},
+		},
+		{
+			name:        "Bool",
+			inputArgs:   []driver.Value{true, false},
+			expectedErr: nil,
+			expected:    []*string{aws.String("1"), aws.String("0")},
+		},
+		{
+			name:        "Zero-value time",
+			inputArgs:   []driver.Value{time.Time{}},
+			expectedErr: nil,
+			expected:    []*string{aws.String("'0000-00-00'")},
+		},
+		{
+			name:        "501 nanoseconds is still < 1 microsecond", // From TestConnection_InterpolateParams_Bool
+			inputArgs:   []driver.Value{time.Time{}.Add(1 * time.Nanosecond)},
```

Comment:
Nit; the `1` here is probably not making any difference?
although, i'm still working out how the expected result is `0001-01-01` and not just zeros ...

Reply:
yep, that's correct

Reply:
Ah yeah, it's not obvious from the test, but the `buildExecutionParams` function adds an additional 500 nanoseconds (copied this behavior from the existing `interpolateParams` function, to maintain the same behavior as before). 

I believe `0001-01-01` is the normal zero-value `time.Time`, so that's why it's like that here.

As for why it's `0000-00-00` in the other test case and not here, `buildExecutionParams`, copying the behavior from `interpolateParams`, special-cases the zero-value `time.Time` and specifically returns `0000-00-00` instead.

Reply:
i see. 👍

Reply:
Removed the unnecessary `1 *` from `1 * time.Nanosecond`.